src/runDiff.ts

```diff
@@ -0,0 +1,225 @@
+      outputFilename: core.getInput("output_filename", { required: true }),
+      ref: core.getInput("ref", { required: true }),
+      stack: core.getInput("stack", { required: true }),
+      stubOutputFile: core.getInput("stub_output_file"),
+      terraformVersion: core.getInput("terraform_version") || "1.8.0",
+      workingDirectory: core.getInput("working_directory") || "./",
+      skipSynth: core.getBooleanInput("skip_synth"),
+      artifactName: core.getInput("artifact_name"),
+    };
+
+    // Then create octokit with the token
+    this.octokit = github.getOctokit(this.inputs.githubToken);
+    this.context = github.context;
+    core.debug("Context:");
+    core.debug(JSON.stringify(this.context));
+  }
+
+  /**
+   * Main execution method. Gets the job ID, runs the CDKTF diff, and processes the results.
+   * Sets action outputs and writes results to a file. Fails the action if errors are detected.
+   */
+  async run() {
+    // Get job information
+    const { job_id, html_url } = await this.getJobId();
+
+    // Run the diff
+    const { result_code, summary } = await this.runDiff();
+
+    // Prepare outputs
+    const outputs: ActionOutputs = {
+      html_url,
+      job_id: job_id.toString(),
+      result_code,
+      stack: this.inputs.stack,
+      summary
+    };
+
+    // Set action outputs
+    Object.entries(outputs).forEach(([key, value]) => {
+      core.setOutput(key, value);
+    });
+
+    // Write outputs to file
+    const outputPath = path.join(this.inputs.workingDirectory, this.inputs.outputFilename);
+    fs.writeFileSync(outputPath, JSON.stringify(outputs));
+
+    if(result_code === "1") {
+      core.setFailed(summary);
+    }
+  }
+
+  /**
+   * Retrieves the job ID and HTML URL for the current workflow job.
+   * 
+   * @returns Promise containing the job ID and HTML URL
+   * @throws Error if the job cannot be found
+   */
+  async getJobId(): Promise<{ job_id: number; html_url: string }> {
+    const octokitPaginatedJobs = await this.octokit.paginate(
+      this.octokit.rest.actions.listJobsForWorkflowRun,
+      {
+        ...github.context.repo,
+        run_id: github.context.runId
+      }
+    );
+
+    const job = octokitPaginatedJobs.find(j => j.name === this.inputs.jobName);
+
+    if (job) {
+      return {
+        job_id: job.id,
+        html_url: job.html_url || ""
+      };
+    } else {
+      throw new Error(`Could not find job with name ${this.inputs.jobName}`);
+    }
+  }
+
+  /**
+   * Executes the CDKTF diff command and processes its output.
+   * Supports both real execution and test mode with stub output.
+   * 
+   * @returns Promise containing result code and summary
+   * @throws Error if the diff execution fails unexpectedly
+   */
+  async runDiff(): Promise<{ result_code: ActionOutputs["result_code"]; summary: string }> {
+    const outputPath = path.join(os.tmpdir() || "/tmp", "cdktf-diff.txt");
+    let diffCommand = this.inputs.stubOutputFile ? 
```

Comment:
Just a thought on this `diffCommand` variable: rather than declaring it as a string and then concatenating to it (making sure you remember leading spaces), what if we just declared it as an array:

```typescript
let diffCommand = [this.inputs.stubOutputFile ? `cat ${this.inputs.stubOutputFile}`: "CI=1 npx cdktf diff"];
if (this.inputs.skipSynth) diffCommand.push(" --skip-synth");
diffCommand.push(this.inputs.stack);

try {
      let output = "";
      await exec.exec("bash", ["-c", diffCommand.join(' ')], {
     // ...
```

This is often considered good practice for memory reason, but this use-case is small enough that I don't think it's relevant. However, I do think it's a little more maintainable long-term.